### PR TITLE
Fix `compile_executable` tests and builld python3.10 from source in linux-arm64 script

### PR DESCRIPTION
**Pull request by tzunghanjuang**
**Context:**

- Delete files generated by the executable tests. 
- Add more possible paths for searching `libpython3`.
- Also try to test if python 3.10 wheel works (Linux Arm64). In the original version, we do not install python-dev-3.10 and it causes troubles for the linker. Hope restoring python-dev-3.10 does not trigger errors.

**Related GitHub Issues:**
#1046

### CODE REVIEW

File: .github/workflows/scripts/linux_arm64/rh8/test_wheels.sh [hunk 1]
```diff
@@ -12,16 +12,8 @@ export PYTHON_PACKAGE=$4
 # Install system dependencies (gcc gives access to c99, which is needed by some tests)
 dnf update -y 
 dnf install -y libzstd-devel gcc-toolset-${GCC_VERSION} gcc
-if [ "$PYTHON_VERSION" != "3.10" ]; then
-    dnf install -y ${PYTHON_PACKAGE} ${PYTHON_PACKAGE}-devel
-fi
```
Comment (dime10) [inline .github/workflows/scripts/linux_arm64/rh8/test_wheels.sh:17]: Is this going to work? I thought we might want to test removing the `if` guard 😅

Comment (tzunghanjuang) [reply .github/workflows/scripts/linux_arm64/rh8/test_wheels.sh:17]: My bad. I deleted the installation part by mistake.

**Review by dime10 — COMMENTED** [review]
Thanks @tzunghanjuang, looks good to me :)

**Review by rauletorresc — CHANGES_REQUESTED** [review]
Could you please replace `${PYTHON_VERSION}` with `${PYTHON_MAJOR_MINOR}` and `${PYTHON_SUBVERSION}` with `${PYTHON_PATCH}`,  I forgot to update this variables in this script. The GitHub workflow already has those new names. Thanks!

File: .github/workflows/scripts/linux_arm64/rh8/test_wheels.sh [hunk 2]
```diff
@@ -14,14 +14,22 @@ dnf update -y
...
+    dnf install openssl-devel bzip2-devel libffi-devel -y
+    dnf install wget -y
+    cd /tmp
+    wget https://www.python.org/ftp/python/${PYTHON_VERSION}.${PYTHON_SUBVERSION}/Python-${PYTHON_VERSION}.${PYTHON_SUBVERSION}.tgz
+    tar xzf Python-${PYTHON_VERSION}.${PYTHON_SUBVERSION}.tgz
+    cd Python-${PYTHON_VERSION}.${PYTHON_SUBVERSION}
+    ./configure --enable-optimizations --enable-shared
+    make
+    make altinstall
+    cd /catalyst
```
[the remark this answers is not in the document]
Comment (tzunghanjuang) [reply .github/workflows/scripts/linux_arm64/rh8/test_wheels.sh:29]: It needs about 20 mins on mac1.

**Comment by rauletorresc** [comment]
> Could you please replace `${PYTHON_VERSION}` with `${PYTHON_MAJOR_MINOR}` and `${PYTHON_SUBVERSION}` with `${PYTHON_PATCH}`, I forgot to update this variables in this script. The GitHub workflow already has those new names. Thanks!

Thanks for the change! But don't forget to propagate the changes to the whole file, there are still parts using the old names...

File: .github/workflows/scripts/linux_arm64/rh8/test_wheels.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (dime10) [reply .github/workflows/scripts/linux_arm64/rh8/test_wheels.sh]: oof

**Comment by rauletorresc** [comment]
> Hi @rauletorresc. Do I still need to keep the `export` for them?

Yes, we are just changing the naming convention

**Review by rauletorresc — APPROVED** [review]
LGTM

File: .github/workflows/scripts/linux_arm64/rh8/test_wheels.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (dime10) [reply .github/workflows/scripts/linux_arm64/rh8/test_wheels.sh]: To be honest that's pretty long but I don't know if we another option at the moment. I tried looking around but if we really need the `libpython3.10.so` then building it ourselves might be the only option. I don't know if using a package built for another distro is an option or not, or we could host our own repository with that package built that is probably too much effort for now.

**Comment by rauletorresc** [comment]
Closes #1046

**Pull request merged**